### PR TITLE
2518: Calendar - Added context menu event and icon tooltips settings

### LINKED ISSUE #2518: Calendar: Add more event callbacks, overflow and a day view
**Describe the bug**
Missing functionalities in calendar component.

**To Reproduce**
Go to http://master-enterprise.demo.design.infor.com/components/calendar/test-specific-month.html

**Expected behavior**

- [x] Day view (Moved to https://github.com/infor-design/enterprise/issues/2780)
- [ ] Hover event
- [x] Show overflow events  (beyond the visible ones) - we will use day view for this or by having events on the right side http://master-enterprise.demo.design.infor.com/components/calendar/example-index.html
- [ ] Right click event callback 

**Version**

- ids-enterprise: v4.19

**Platform**
 - OS Version: Windows 10
 - Browser Name: Chrome
 - Browser Version: 75.0.3770.142 (Official Build) (64-bit)

**Additional context**

### PR DESCRIPTION
**Pull request by vonnyw**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Adds support for context menu, event tooltip and icon tooltip.

**Related github/jira issue (required)**:
Fixes #2518
See also https://github.com/infor-design/enterprise/compare/SOHO-2518

**Steps necessary to review your pull request (required)**:
1. Go to http://localhost:4000/components/calendar/test-ajax-events.html
2. Click 'Today' 
3. Hover over a calendar event on September 13th, 2019.
4. Hover over a calendar event icon on September 13th, 2019.
5. Right click on a calendar event - a message will be printed on the console

image

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 ### v4.21.0 Features
 
+- `[Calendar]` Added the ability to show tooltip on event and event icon. Also added ability to show event context menu ([#2518](https://github.com/infor-design/enterprise/issues/2518))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md]:
```suggestion
- `[Calendar]` Added the ability to show tooltip on event and event icon and the ability to fire a context menu event. ([#2518](https://github.com/infor-design/enterprise/issues/2518))
```

File: src/components/calendar/calendar.js [hunk 2]
```diff
@@ -24,6 +24,8 @@ const COMPONENT_NAME_DEFAULTS = {
   modalTemplate: null,
   menuId: null,
   menuSelected: null,
+  eventTooltip: 'overflow',
```
Comment (tmcconechy) [inline src/components/calendar/calendar.js]: A little further up in this file is another copy of the settings in the docs. Please add a comment up there so this goes into the documenation thats generated.

**Comment by vonnyw** [comment]
The example is updated with this PR to show the tooltips and context menu.

**Comment by tmcconechy** [comment]
Note that you have a couple of line errors (`npm run lint:ci`) https://travis-ci.com/infor-design/enterprise/jobs/239135889#L523

And some tests appear to fail https://travis-ci.com/infor-design/enterprise/jobs/239135887#L2418

For the tests try to sync your branch with master to fix. (`git pull origin master` usually)

**Comment by vonnyw** [comment]
I updated the test, fixed lint errors and added more details on the steps to review along with print screen.

**Comment by vonnyw** [comment]
Thanks @deep7102. I see the contextmenu event is triggered twice too, once for 'calendar-event' and the other for 'calendar-event-content'. @tmcconechy any suggestion on how not to trigger it for the calendar-event-content?

**Comment by deep7102** [comment]
Hi @vonnyw by adding `e.stopPropagation();` after line below should stop bubbling the event.
```
this.element.off(`contextmenu.${COMPONENT_NAME}`).on(`contextmenu.${COMPONENT_NAME}`, '.calendar-event', (e) => {
```
 https://github.com/infor-design/enterprise/blob/c442a64e150304c6321e81ba6673171c504f423b/src/components/calendar/calendar.js#L700

**Comment by vonnyw** [comment]
Thanks for the suggestion @deep7102 . I made the update as suggested and retested, now the contextmenu event is only triggered once.

**Comment by tmcconechy** [comment]
@vonnyw try merging master onto this branch i see some tests i think we may have fixed here https://travis-ci.com/infor-design/enterprise/jobs/239975856

**Comment by EdwardCoyle** [comment]
Seems there are some consistently-failing validation tests?  Would those also be failing from `master`?

Before the last `master` merge:  https://travis-ci.com/infor-design/enterprise/jobs/239975856#L2284
After: https://travis-ci.com/infor-design/enterprise/jobs/240342688#L3010

Maybe the test just needs adjustment.

**Comment by tmcconechy** [comment]
I'll take a look at that now - running the tests on my VM. The one is new the other has been problematic.

**Comment by vonnyw** [comment]
Please let me know if my PR caused this test failure, it shouldn't affect validation component.

**Pull request merged**